openstudiocore/src/model/test/ZonePropertyUserViewFactorsBySurfaceName_GTest.cpp

```diff
@@ -0,0 +1,303 @@
+*
+*  (4) Other than as required in clauses (1) and (2), distributions in any form of modifications or other derivative works
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+#include <string>
+
+#include "ModelFixture.hpp"
+
+#include "../Model.hpp"
+#include "../Model_Impl.hpp"
+
+#include "../ZonePropertyUserViewFactorsBySurfaceName.hpp"
+#include "../ZonePropertyUserViewFactorsBySurfaceName_Impl.hpp"
+
+#include "../ThermalZone.hpp"
+#include "../ThermalZone_Impl.hpp"
+#include "../Surface.hpp"
+#include "../Surface_Impl.hpp"
+#include "../SubSurface.hpp"
+#include "../SubSurface_Impl.hpp"
+#include "../InternalMass.hpp"
+#include "../InternalMass_Impl.hpp"
+#include "../InternalMassDefinition.hpp"
+#include "../InternalMassDefinition_Impl.hpp"
+#include "../Space.hpp"
+
+using namespace openstudio;
+using namespace openstudio::model;
+
+
+TEST_F(ModelFixture, ZonePropertyUserViewFactorsBySurfaceName_ViewFactor) {
+
+  Model m;
+  Point3dVector points;
+  points.clear();
+  points.push_back(Point3d(0, 2, 0));
+  points.push_back(Point3d(0, 0, 0));
+  points.push_back(Point3d(1, 0, 0));
+  Surface s(points, m);
+
+  InternalMassDefinition intMassDefn(m);
+  intMassDefn.setSurfaceArea(20);
+  InternalMass intMass(intMassDefn);
+
+  ThermalZone z(m);
+
+  // OK
+  EXPECT_NO_THROW(ViewFactor(s, intMass, 0.9));
+
+  // Wrong double
+  EXPECT_THROW(ViewFactor(s, intMass, 1.2), openstudio::Exception);
+
+  // Wrong type
+  EXPECT_THROW(ViewFactor(s, z, 0.9), openstudio::Exception);
+  EXPECT_THROW(ViewFactor(z, s, 0.9), openstudio::Exception);
+
+
+  // TODO: JM 2019-09-13 NOT SURE WHETHER WE WANT TO ALLOW TO==FROM here
+  // Test that you cannot add a view factor  if toSurface == fromSurface
+  EXPECT_THROW(ViewFactor(s, s, 0.25),  openstudio::Exception);
```

Comment:
Will want to change this to no throw.

Reply:
It looks like i forgot to push my last commit... damn! This is on work machine that's off and i dont go back until Monday